### PR TITLE
CLOUDSTACK-8556: Unable to delete attached volume in cleanup

### PR DESCRIPTION
**Pull request by pritisarap12**

### CODE REVIEW

File: test/integration/testpaths/testpath_snapshot_limits.py [hunk 1]
```diff
@@ -341,4 +339,8 @@ def test_05_storage_snapshots_limits(self):
             data_volumes_list[0]
         )
 
+	self.vm.reboot(self.apiclient)
```
Comment (gauravaradhye) [inline test/integration/testpaths/testpath_snapshot_limits.py]: Why reboot and then delete? Also add suitable description and test run log results to the description.

**Comment by pritisarap12** [comment]
Removed reboot as it is not needed.
Below is the test result after removing reboot function:

Storage and Snapshot Limit ... === TestName: test_01_storage_snapshots_limits | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 300.585s

OK

**Comment by gauravaradhye** [comment]
Can you please explain why the test case was failing before? Why the cleanup was failing for volume?

**Comment by pritisarap12** [comment]
Delete volume was failing in cleanup before because volume was not in detached state as snapshot creation failed and detach volume command did not get executed so volume remains in attached state.
Now the detach volume is executed in tearDown class itself so clean up does not fail for volume.

File: test/integration/testpaths/testpath_snapshot_limits.py [hunk 2]
```diff
@@ -115,32 +115,51 @@ def tearDownClass(cls):
...
 
     def setUp(self):
+        if not self.snapshotSupported:
+            self.skipTest(
+                "Snapshots are not supported on %s" %
+                self.hypervisor)
+
         self.apiclient = self.testClient.getApiClient()
         self.dbclient = self.testClient.getDbConnection()
         self.cleanup = []
```
Comment (gauravaradhye) [inline test/integration/testpaths/testpath_snapshot_limits.py]: cleanup has to be declared before skipping the test case

Comment (pritisarap12) [inline test/integration/testpaths/testpath_snapshot_limits.py]: Updated testpath as per the review comment.

**Comment by wido** [comment]
The description of the pull request isn't really specific. Looking at the files changed it is only a test case which changed, right? No CS code itself.

**Comment by wido** [comment]
In that case LGTM

Next time try to be more descriptive, since the Jira issue didn't show it either. I really didn't know what you meant.

**Comment by bhaisaab** [comment]
LGTM, squashing and merging; please squash your changes in future as you iterate on your PR

**Pull request closed without merging**